docs/docs/development/local-toolchain/build-flash.mdx

```diff
@@ -3,48 +3,115 @@ title: Building and Flashing
+
+    Given the following:
+
+    - Keyboard: Planck (rev6)
+    - Keymap: default
 
-### Keyboard (Shield) + MCU Board
+    you can build ZMK with the following:
 
-ZMK treats keyboards that take an MCU addon board as [shields](https://docs.zephyrproject.org/3.5.0/hardware/porting/shields.html), and treats the smaller MCU board as the true [board](https://docs.zephyrproject.org/3.5.0/hardware/porting/board_porting.html)
+    ```sh
+    west build -b planck_rev6
+    ```
 
-Given the following:
+  </TabItem>
+  <TabItem value="addonMcu">
+    ZMK treats keyboards that take an MCU addon board as
+    [shields](https://docs.zephyrproject.org/3.5.0/hardware/porting/shields.html),
+    and treats the smaller MCU board as the true
+    [board](https://docs.zephyrproject.org/3.5.0/hardware/porting/board_porting.html).
 
-- MCU Board: Proton-C
-- Keyboard PCB: kyria_left
-- Keymap: default
+    Given the following:
 
-You can build ZMK with the following:
+    - MCU Board: Proton-C
+    - Keyboard PCB: kyria_left
+    - Keymap: default
+
+    You can build ZMK with the following:
+
+    ```sh
+    west build -b proton_c -- -DSHIELD=kyria_left
+    ```
+
+  </TabItem>
+</Tabs>
+
+### CMake Arguments
+
+For creating a build system, the `west build` command uses
+[CMake](https://cmake.org/). This allows for additional arguments to be added to
+its invocations.
+
+The simplest of doing this is by adding `--` after the `west build` command and
+listing the CMake arguments afterward. For example, the above-shown command for
+building a keyboard with an MCU board add-on uses them in the form of
+`-DSHIELD=kyra_left`.
+
+:::tip
+Once the first generation of the build directory using one-time CMake
+arguments is done, you can execute subsequent builds by using:
 
 ```sh
-west build -b proton_c -- -DSHIELD=kyria_left
+west build -d <build_dir>
 ```
 
-### Keyboard With Onboard MCU
-
-Keyboards with onboard MCU chips are simply treated as the [board](https://docs.zephyrproject.org/3.5.0/hardware/porting/board_porting.html) as far as Zephyr™ is concerned.
+:::
 
-Given the following:
+Furthermore, it is possible to configure `west build` by using
+[`west config`](https://docs.zephyrproject.org/3.5.0/develop/west/config.html#west-config-cmd).
+This makes it possible to add permanent CMake arguments, which are invoked
+whenever a new build system is generated. To do this, we make use of its
+`build.cmake-args` configuration option.
 
-- Keyboard: Planck (rev6)
-- Keymap: default
+```sh
+west config build.cmake-args -- -DSHIELD=kyra_left
+```
 
-you can build ZMK with the following:
+For adding multiple arguments a string can be used:
 
 ```sh
-west build -b planck_rev6
+west config build.cmake-args \
+  -- "-DSHIELD=kyra_left -DZMK_CONFIG=/absolute/path/to/zmk-config"
 ```
 
+:::note
+New build systems are only generated with `west build` when they are
+currently is non-available. Therefore, either delete the build directory or
+do a **pristine build**.
+:::
+
```

Comment:
```suggestion
```
I don't think this note is necessary as pristine building is discussed immediately afterwards.